components/navigable-container/index.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * External Dependencies
+ */
+import { omit, noop } from 'lodash';
+
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+import { focus, keycodes } from '@wordpress/utils';
+
+/**
+ * Module Constants
+ */
+const { UP, DOWN, LEFT, RIGHT, TAB } = keycodes;
+
+function cycleValue( value, total, offset ) {
+	const nextValue = value + offset;
+	if ( nextValue < 0 ) {
+		return total + nextValue;
+	} else if ( nextValue >= total ) {
+		return nextValue - total;
+	}
+
+	return nextValue;
+}
+
+class NavigableContainer extends Component {
+	constructor() {
+		super( ...arguments );
+		this.bindContainer = this.bindContainer.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+
+		this.getFocusableContext = this.getFocusableContext.bind( this );
+		this.getFocusableIndex = this.getFocusableIndex.bind( this );
+	}
+
+	bindContainer( ref ) {
+		this.container = ref;
+	}
+
+	getFocusableContext( target ) {
+		const { deep = false, onlyBrowserTabstops } = this.props;
+		const finder = onlyBrowserTabstops ? focus.tabbable : focus.focusable;
+		const focusables = finder
+			.find( this.container )
+			.filter( ( node ) => deep || node.parentElement === this.container );
+
+		const index = this.getFocusableIndex( focusables, target );
+		if ( index > -1 && target ) {
+			return { index, target, focusables };
+		}
+		return null;
+	}
+
+	getFocusableIndex( focusables, target ) {
+		const directIndex = focusables.indexOf( target );
+		if ( directIndex !== -1 ) {
+			return directIndex;
+		}
+	}
+
+	onKeyDown( event ) {
+		if ( this.props.onKeyDown ) {
+			this.props.onKeyDown( event );
+		}
+
+		const { getFocusableContext } = this;
+		const { cycle = true, eventToOffset, onNavigate = noop, stopNavigationEvents } = this.props;
+
+		const offset = eventToOffset( event );
+
+		// eventToOffset returns undefined if the event is not handled by the component
+		if ( offset !== undefined && stopNavigationEvents ) {
+			// Prevents arrow key handlers bound to the document directly interfering
+			event.nativeEvent.stopImmediatePropagation();
+			event.preventDefault();
+			event.stopPropagation();
+		}
+
+		if ( ! offset ) {
+			return;
+		}
+
+		const context = getFocusableContext( document.activeElement );
+		if ( ! context ) {
+			return;
+		}
+
+		const { index, focusables } = context;
+		const nextIndex = cycle ? cycleValue( index, focusables.length, offset ) : index + offset;
+		if ( nextIndex >= 0 && nextIndex < focusables.length ) {
+			focusables[ nextIndex ].focus();
+			onNavigate( nextIndex, focusables[ nextIndex ] );
+		}
+	}
+
+	render() {
+		const { children, ...props } = this.props;
+
+		// Disable reason: Assumed role is applied by parent via props spread.
+		/* eslint-disable jsx-a11y/no-static-element-interactions */
```

Comment:
This rule is disabled but then never turned on again (meaning it will take effect for the remainder of the file).